lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
+
+  def initialize(noid, config = nil, application_logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = application_logger || PushmiPullyu.logger
+    @fetcher = PushmiPullyu::FedoraObjectFetcher.new(self.noid, self.config)
+
+    initialize_directory_names
+    initialize_filenames
+
+    # We need to log to application log and to the AIP log
+    @aip_logger = PushmiPullyu::AipLogger.new(@noid, aipcreation_log, @logger)
```

Comment:
My hangup is that we're approaching logging in different ways in different places for no good reason. There's not even consistency with regards to formatting between the two logging approaches.

> While it's true that multiple threads could write to the same application logger, I don't think two threads would write to the AIP creation log.

How so? If two items are being preserved at roughly the same time, they would both be writing to the log in an indeterminate order. You're not creating a log per-object are you? The planned  approach was that we would log preservation actions to a single common preservation log, separate from the general application log, with an eye towards moving the preservation log elsewhere at a later date, and the Librarians grepping the common log for a noid of interest for the time being.